projects/ng-dynamic-component/src/lib/io/io.service.ts

```diff
@@ -298,7 +298,11 @@ export class IoService implements OnDestroy {
       const outputExpr = outputs[key];
 
       if (typeof outputExpr === 'function') {
-        processedOutputs[key] = outputExpr;
+        processedOutputs[key] = event => {
+          // Mark host component for check every time the event handler is called
+          this.cdr?.markForCheck();
```

Comment:
Instead of having CDR checks in 2 places you can move it into single place in the `bindOutputs()` method where the subscriptions are created.
Also you should first try to use dynamic component's CDR (available in `compCdr` prop) and if not available then fallback to it's own injected CDR.

Reply:
moved it to `bindOutputs`.

I tried using `compCdr`, but that didn't work. Always using the host's ChangeDetectorRef is also more correct, because the injected component does not need another change detection. Only the component that handles the event needs to be change detected again.

Reply:
Sounds logical, I just thought that marking the dynamic component itself should yield the same result as marking parent - because it's located under parent in the tree so it should become marked as every other parent component...
Don't quite understand why it does not mark parent component.
Maybe you know the reason?